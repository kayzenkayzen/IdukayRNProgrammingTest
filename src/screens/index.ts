export { default as PotionsSelectionScreenContainer } from '@src/screens/PotionsSelectionScreen/PotionsSelectionScreenContainer';
export { default as PotionsSelectionScreen } from '@src/screens/PotionsSelectionScreen/PotionsSelectionScreen';
export { default as OptimalDamageScreenContainer } from '@src/screens/OptimalDamageScreen/OptimalDamageScreenContainer';
export { default as OptimalDamageScreen } from '@src/screens/OptimalDamageScreen/OptimalDamageScreen';
