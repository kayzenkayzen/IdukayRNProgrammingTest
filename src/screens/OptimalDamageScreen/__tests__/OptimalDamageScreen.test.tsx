import React from 'react';
import { render } from '@testing-library/react-native';
import { OptimalDamageScreen } from '@src/screens';

describe('Test OptimalDamageScreen Component', () => {
  it('renders correctly', () => {
    const route = { params: {} };

    const wrapper = render(<OptimalDamageScreen route={route} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('renders attacks correctly', () => {
    const route = {
      params: {
        attacks: [['red', 'blue', 'green'], ['red', 'blue', 'green'], ['red']],
      },
    };

    const { getByTestId } = render(<OptimalDamageScreen route={route} />);

    expect(getByTestId('attack-0')).toBeDefined();
    expect(getByTestId('attack-1')).toBeDefined();
    expect(getByTestId('attack-2')).toBeDefined();
    expect(getByTestId('attack-0-red-0')).toBeDefined();
    expect(getByTestId('attack-0-blue-1')).toBeDefined();
    expect(getByTestId('attack-0-green-2')).toBeDefined();
    expect(getByTestId('attack-1-red-0')).toBeDefined();
    expect(getByTestId('attack-1-blue-1')).toBeDefined();
    expect(getByTestId('attack-1-green-2')).toBeDefined();
    expect(getByTestId('attack-2-red-0')).toBeDefined();
  });

  it('renders damage correctly', () => {
    const route = {
      params: {
        damage: 15,
      },
    };

    const { getByTestId } = render(<OptimalDamageScreen route={route} />);

    expect(getByTestId('damage-value').props.children).toBe('15 %');
  });
});
