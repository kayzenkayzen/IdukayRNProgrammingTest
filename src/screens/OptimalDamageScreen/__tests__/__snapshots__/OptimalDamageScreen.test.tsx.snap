// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test OptimalDamageScreen Component renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2b2b2b",
              "flex": 1,
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 16,
              "lineHeight": 21,
            }
          }
        >
          Realiza estos ataques para lograr el mayor da√±o a tu objetivo
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "green",
              "borderRadius": 75,
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 20,
              "width": 75,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 70,
                "height": 70,
                "justifyContent": "center",
                "width": 70,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "green",
                  "fontFamily": "SourceSansPro-Bold",
                  "fontSize": 21,
                  "textAlign": "center",
                }
              }
              testID="damage-value"
            >
              undefined %
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
